File: average_calculator.ts
// TypeScript program to calculate the average of four numbers entered by the user

import * as readline from 'readline';

function main(): void {
    const rl = readline.createInterface({
        input: process.stdin,
        output: process.stdout
    });

    console.log("Enter four numbers to calculate their average:");

    const numbers: number[] = [];
    let count = 0;

    function ask(): void {
        if (count < 4) {
            rl.question(`Number ${count + 1}: `, (answer: string) => {
                const num = parseFloat(answer);
                if (isNaN(num)) {
                    console.log("Invalid input. Please enter a valid number.");
                    ask();
                } else {
                    numbers.push(num);
                    count++;
                    ask();
                }
            });
        } else {
            rl.close();
            const average = numbers.reduce((a, b) => a + b, 0) / 4;
            console.log(`The average of ${numbers.join(', ')} is: ${average.toFixed(2)}`);
        }
    }

    ask();
}

main();

File: average_calculator.py
# Program to calculate the average of four numbers entered by the user

def main():
    print("Enter four numbers to calculate their average:")
    
    try:
        # Get four numbers from user input
        num1 = float(input("First number: "))
        num2 = float(input("Second number: "))
        num3 = float(input("Third number: "))
        num4 = float(input("Fourth number: "))
        
        # Calculate the average
        average = (num1 + num2 + num3 + num4) / 4
        
        # Display the result
        print(f"The average of {num1}, {num2}, {num3}, and {num4} is: {average:.2f}")
    
    except ValueError:
        print("Invalid input. Please enter valid numbers.")

if __name__ == "__main__":
    main()